packages/open-truss/src/signals/index.ts

```diff
@@ -21,19 +36,25 @@ export function getSignalsType(
   }
 }
 
+function isSignalLike(obj: unknown): obj is Signal {
+  return typeof obj === 'object' && obj !== null && 'value' in obj
+}
+
 export function SignalType<T>(
   name: string,
-  defaultValue: T,
-  validator?: () => boolean,
-): z.ZodDefault<z.ZodType<Signal<T>>> {
-  validator ||= () => true
+  valueShape: ValueShape<T>,
+): SignalsZodType<T> {
+  const defaultValue = valueShape.parse(undefined)
+  const validator = (val: unknown): boolean => {
+    return isSignalLike(val) && valueShape.parse(val.value) !== undefined
```

Comment:
@kmcq Jon and I were reviewing this and were wondering if `valueShape.parse(val.value)` could ever be `undefined`?

Reply:
Hmm I don't remember why I did this. I think we could just do this:

```suggestion
    return isSignalLike(val) && valueShape.parse(val.value)
```